x-pack/elastic-agent/pkg/agent/application/pipeline/emitter/modifiers/fleet_decorator.go

```diff
@@ -39,7 +40,7 @@ func InjectFleet(cfg *config.Config, hostInfo types.HostInfo, agentInfo *info.Ag
 		// copy top-level agent.* into fleet.agent.* (this gets sent to Applications in this structure)
 		if agent, ok := transpiler.Lookup(ast, "agent"); ok {
 			if err := transpiler.Insert(ast, agent, "fleet"); err != nil {
-				return err
+				return errors.New("inserting agent info failed", err)
```

Comment:
Is there a reason, that above you put the existing error first and here as second argument?

Reply:
nope, order does not matter here